services/src/transaction_service.rs

```diff
@@ -0,0 +1,216 @@
+                    .execute(tx_recv, notifier.clone(), exit_signal_t.clone());
+
+            let replay_service = tx_replayer.start_service(
+                replay_channel_task,
+                replay_reciever,
+                exit_signal_t.clone(),
+            );
+
+            let finalized_block_listener = block_listner.clone().listen(
+                CommitmentConfig::finalized(),
+                notifier.clone(),
+                tpu_service.get_estimated_slot_holder(),
+                exit_signal_t.clone(),
+            );
+
+            let confirmed_block_listener = block_listner.clone().listen(
+                CommitmentConfig::confirmed(),
+                None,
+                tpu_service.get_estimated_slot_holder(),
+                exit_signal_t.clone(),
+            );
+
+            let processed_block_listener = block_listner
+                .clone()
+                .listen_processed(exit_signal_t.clone());
+
+            let cleaner = Cleaner::new(tx_sender.clone(), block_listner.clone(), block_store_t)
+                .start(clean_interval, exit_signal_t);
+
+            tokio::select! {
+                res = tpu_service_fx => {
+                    format!("{res:?}")
+                },
+                res = tx_sender_jh => {
+                    format!("{res:?}")
+                },
+
+                res = finalized_block_listener => {
+                    format!("{res:?}")
+                },
+                res = confirmed_block_listener => {
+                    format!("{res:?}")
+                },
+                res = processed_block_listener => {
+                    format!("{res:?}")
+                },
+                res = replay_service => {
+                    format!("{res:?}")
+                },
+                res = cleaner => {
+                    format!("{res:?}")
+                },
+            }
+        });
+
+        (
+            TransactionService {
+                transaction_channel,
+                replay_channel,
+                exit_signal,
+                block_store,
+                max_retries,
+                replay_after: self.tx_replayer.retry_after,
+            },
+            jh_services,
+        )
+    }
+}
+
+#[derive(Clone)]
+pub struct TransactionService {
+    pub transaction_channel: Sender<(String, WireTransaction, u64)>,
+    pub replay_channel: UnboundedSender<TransactionReplay>,
+    pub exit_signal: Arc<AtomicBool>,
+    pub block_store: BlockStore,
+    pub max_retries: usize,
+    pub replay_after: Duration,
+}
+
+impl TransactionService {
+    pub async fn send_transaction(
+        &self,
+        raw_tx: Vec<u8>,
+        max_retries: Option<u16>,
+    ) -> anyhow::Result<String> {
+        let tx = match bincode::deserialize::<VersionedTransaction>(&raw_tx) {
+            Ok(tx) => tx,
+            Err(err) => {
+                bail!(err.to_string());
+            }
+        };
+        let signature = tx.signatures[0];
```

Comment:
maybe rename to sth more specific ("signer")

Reply:
it is actually the signature for the transaction actually.